packages/server/src/auth/CrossmintAuth.ts

```diff
@@ -31,16 +32,28 @@ export class CrossmintAuth {
         return new CrossmintAuth(crossmint);
     }
 
-    public async getSession(options: GenericRequest | AuthMaterialBasic): Promise<AuthSession> {
+    public async getSession(
+        options: GenericRequest | AuthMaterialBasic,
+        response?: GenericResponse
+    ): Promise<AuthSession> {
         const { jwt, refreshToken } = "refreshToken" in options ? options : getAuthCookies(options);
 
         if (!refreshToken) {
             throw new CrossmintAuthenticationError("Refresh token not found");
         }
 
         try {
-            return await this.validateOrRefreshSession(jwt, refreshToken);
+            return await this.validateOrRefreshSession(jwt, refreshToken, response);
         } catch (error) {
+            if (error instanceof CrossmintAuthenticationError && response != null) {
+                this.storeAuthMaterial(response, {
+                    jwt: "",
+                    refreshToken: {
```

Comment:
I've added the `refreshToken` so it can be stored at will. It follows the object format from the response from `/refresh` to use the expiry date when configuring the cookie. An alternative would be to just pass around the string (simpler code) and set the cookie without an expiry and once it's invalid, it will throw errors anyway